### ISSUE TITLE
Cannot fetch proxy status on Istio master (1.11)

### ISSUE THREAD

**Issue by howardjohn** [kiali/kiali#4132]
**Describe the bug**
```
{"error":"Error fetching the proxy-status in the following pods: istiod-5648fdc58f-shp8q: the server has asked for the client to provide credentials (get pods istiod-5648fdc58f-shp8q)"}{"errors":0,"objectCount":0,"warnings":0}
```

This is almost certainly from changes to add authentication to the Istiod debug interface. We should work together to see the best path forward on this. PR: https://github.com/istio/istio/pull/31338

For UI issues please also add a screenshot that shows the issue.

**Versions used**
Kiali: 1.36
Istio: master (1.11)
Kubernetes flavour and version: (e.g. OpenShift Origin 3.9) kind 1.22

**To Reproduce**
1. Install Istio from master with Kiali 1.36. 
2. Access some page like `http://localhost:20001/kiali/api/namespaces/local-path-storage/validations`

**Expected behavior**
API doesn't fail with error

**Comment by lucasponce** [comment]
It sounds similar to https://github.com/kiali/kiali/issues/4054#issuecomment-869856601 or perhaps it's a new one ?

cc @xeviknal

**Comment by howardjohn** [comment]
Yes, looks like the same issue. We can merge with that one. Let me know if I can help out here at all

**Comment by howardjohn** [comment]
Actually I am not 100% sure. https://github.com/kiali/kiali/pull/4108 definitely would not fix this I think. You need to either include a SA token in the request or come from localhost (basically only possible with port-forward)

**Comment by lucasponce** [comment]
> Yes, looks like the same issue. We can merge with that one. Let me know if I can help out here at all

Well, we're already @xeviknal and I discussing how to access to this information programatically, what could be the requirements from Kiali to access to that endpoint with the new architecture ?

**Comment by lucasponce** [comment]
Yes #4108 is a draft where we're brainstorming several options, probably we think we'd keep the portforward technique to access to the IstioD enpoints but yes, we don't want to reinvent the wheel, if from Istio there is some establish way to do it, it just to adapt it on Kiali.

**Comment by howardjohn** [comment]
Basically equivalent to:

```
  echo '{"kind":"TokenRequest","apiVersion":"authentication.k8s.io/v1","spec":{"audiences":["istio-ca"], "expirationSeconds":604800}}' | kubectl create --raw /api/v1/namespaces/NAMESPACE/serviceaccounts/SERVICE_ACCOUNT/token -f - | jq -j '.status.token' > /tmp/token
        curl https://istiod-ip:8080/debug/SOME_URL -H "authorization: Bearer $(cat /tmp/token)" -s
```

You can use a SA token mounted on the deployment as well which is probably better.

**Comment by lucasponce** [comment]
Thanks @howardjohn.

@xeviknal I think the #4108 should be used also to cover this one.

**Comment by xeviknal** [comment]
Closing this one. Fixed by #4108
